common/djangoapps/third_party_auth/models.py

```diff
@@ -140,6 +140,13 @@ class ProviderConfig(ConfigurationModel):
             "authentication using the correct link is still possible."
         ),
     )
+    drop_existing_session = models.BooleanField(
+        default=False,
+        help_text=_(
+            "Whether to drop an existing session when accessing certain views and including "
```

Comment:
I'm not a huge fan of the ambiguous "certain views" here, which makes this all but useless except to those select few who already know exactly what this does.

What do you think about just making your tpa_hint_ends_existing_session code into a tpa_hint middleware, so that it applies consistently to all views? Since the middleware would only do anything if request.GET contains tpa_hint it would not really add any overhead, so I don't think there's any efficiency concerns, and that approach would be much easier to understand and reason about.